File: src/types/events.ts
export type PetEvent = 
  | { type: "TimeTick" }
  | { type: "UserInteract"; action: UserAction }
  | { type: "SystemEvent"; info: SystemInfo }
  | { type: "ReminderEvent"; reminder: Reminder }
  | { type: "AiDecision"; action: AiAction };

export type UserAction = 
  | { type: "Feed" }
  | { type: "Play" }
  | { type: "Pet" }
  | { type: "Talk" }
  | { type: "Command"; command: string };

export interface SystemInfo {
  cpu_usage: number;
  memory_usage: number;
  network_activity: boolean;
}

export interface Reminder {
  id: string;
  title: string;
  message: string;
  reminder_type: ReminderType;
}

export type ReminderType = 
  | { type: "Study" }
  | { type: "Break" }
  | { type: "Custom"; name: string };

export type AiAction = 
  | { type: "Wander" }
  | { type: "React"; reaction: string }
  | { type: "Suggest"; suggestion: string };

File: src-tauri/src/modules/state_machine/events.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum PetEvent {
    TimeTick,
    UserInteract(UserAction),
    SystemEvent(SystemInfo),
    ReminderEvent(Reminder),
    AiDecision(AiAction),
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum UserAction {
    Feed,
    Play,
    Pet,
    Talk,
    Command(String),
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SystemInfo {
    pub cpu_usage: f32,
    pub memory_usage: f32,
    pub network_activity: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Reminder {
    pub id: String,
    pub title: String,
    pub message: String,
    pub reminder_type: ReminderType,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ReminderType {
    Study,
    Break,
    Custom(String),
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum AiAction {
    Wander,
    React(String),
    Suggest(String),
}